plugins/modules/external_account.py

```diff
@@ -0,0 +1,381 @@
+        api_instance = ExternalAccountsResourceApi(self.api_client)
+        existing = []
+        self.params = {
+            key: value for key, value in self.params.items() if value is not None
+        }
+        try:
+            existing = api_instance.read_account(self.name).to_dict()
+        except ApiException as ex:
+            if ex.status == 400:
+                pass
+            else:
+                raise ex
+
+        if self.state == "present":
+            try:
+                if existing:
+                    if not self.module.check_mode:
+                        self.external_account_output = api_instance.update_account(
+                            body=ApiExternalAccount(
+                                name=self.name,
+                                display_name=self.name,
+                                type_name=self.type,
+                                account_configs=ApiConfigList(
+                                    items=[
+                                        ApiConfig(name=key, value=value)
+                                        for key, value in self.params.items()
+                                    ]
+                                ),
+                            )
+                        )
+                        self.changed = True
+                else:
+                    if not self.module.check_mode:
+                        self.external_account_output = api_instance.create_account(
+                            body=ApiExternalAccount(
+                                name=self.name,
+                                display_name=self.name,
+                                type_name=self.type,
+                                account_configs=ApiConfigList(
+                                    items=[
+                                        ApiConfig(name=key, value=value)
+                                        for key, value in self.params.items()
+                                    ]
+                                ),
+                            )
+                        )
+                        self.changed = True
+
+            except ApiException as ex:
+                if ex.status == 400:
+                    self.module.fail_json(msg=json.loads(ex.body)["message"])
+                else:
+                    raise ex
+
+        if self.state == "absent":
+            if existing:
+                self.external_account_output = api_instance.delete_account(self.name)
+                self.changed = True
+
+
+def main():
+
+    module = ClouderaManagerModule.ansible_module(
+        argument_spec=dict(
+            name=dict(required=False, type="str"),
+            category=dict(
+                type="str",
+                required=False,
+                choices=["AWS", "ALTUS", "AZURE", "BASICAUTH"],
+            ),
+            state=dict(
+                type="str",
+                default="present",
+                choices=["present", "absent"],
```

Comment:
Need to add logic for `state=present` to ensure that `category` and `type` are set.